compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -634,6 +643,17 @@ tyConAnchor DocCtx{..} tycon = do
             | otherwise = typeAnchor
     Just (anchorFn mod name)
 
+-- | Create a (possibly external) reference from a TyCon.
+tyConReference :: DocCtx -> TyCon -> Maybe Reference
+tyConReference ctx@DocCtx{..} tycon = do
+    referenceAnchor <- tyConAnchor ctx tycon
+    let ghcName = tyConName tycon
+        referencePackage = do
+            guard (not (nameIsHomePackage dc_ghcMod ghcName))
```

Comment:
Does it ever happen that we get `Nothing` here but `nameModule_maybe` returns `Just`? My understanding was that during compilation we have an `IndefiniteUnitId` for the current package so that case should already check for whether the name is from the current package.

Reply:
Hm. It seems to affect the docs somehow (if I add this, the instances disappear from the golden tests ... which seems like a bug. But it does suggest that by the time we look at instances the unit IDs are no longer indefinite.). I need to investigate this further.

Reply:
It could very well just be me misunderstanding unit ids. No need to investigate this before merging or even anytime soon if it works but maybe add a comment.